account_financial_report_qweb/report/journal_report.py

```diff
@@ -0,0 +1,599 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+DIGITS = (16, 2)
+
+
+class ReportJournalQweb(models.TransientModel):
+
+    _name = 'report_journal_qweb'
+
+    date_from = fields.Date()
+    date_to = fields.Date()
+    company_id = fields.Many2one(
+        comodel_name='res.company'
+    )
+    move_target = fields.Selection(
+        selection='_get_move_targets',
+        default='all',
+    )
+    journal_ids = fields.Many2many(
+        comodel_name='account.journal',
+    )
+    report_journal_ids = fields.One2many(
+        comodel_name='report_journal_qweb_journal',
+        inverse_name='report_id',
+    )
+
+    @api.model
+    def _get_move_targets(self):
+        return self.env['journal.report.wizard']._get_move_targets()
+
+    @api.multi
+    def compute_data_for_report(self):
+        self.ensure_one()
+        self._inject_journal_values()
+        self._inject_move_values()
+        self._inject_move_line_values()
+        self._inject_journal_tax_values()
+        self._update_journal_report_total_values()
+
+    @api.multi
+    def refresh(self):
+        self.ensure_one()
+        self.report_journal_ids.unlink()
+        self.compute_data_for_report()
+
+    @api.multi
+    def _inject_journal_values(self):
+        self.ensure_one()
+        sql = """
+            INSERT INTO report_journal_qweb_journal (
+                create_uid,
+                create_date,
+                report_id,
+                journal_id,
+                name,
+                code
+            )
+            SELECT
+                %s as create_uid,
+                NOW() as create_date,
+                %s as report_id,
+                aj.id as journal_id,
+                aj.name as name,
+                aj.code as code
+            FROM
+                account_journal aj
+            WHERE
+                aj.id in %s
+            AND
+                aj.company_id = %s
+            ORDER BY
+                aj.name
+        """
+        params = (
+            self.env.uid,
+            self.id,
+            tuple(self.journal_ids.ids),
```

Comment:
Currenlty, journal_ids isn't required. So, this line will raise an exception.

Reply:
I forgot to add the constrains. Thanks